### PR TITLE
Add Windows compatibility

### PR DESCRIPTION
**Pull request by vderunov**

### CODE REVIEW

File: src/main/follower.ts [hunk 1]
```diff
@@ -28,12 +30,21 @@ export function followerKill() {
...
+    return await getPid('ipfs-cluster-follow synthetix run');
+  }
+
   return await getPid('.synthetix/ipfs-cluster-follow/ipfs-cluster-follow synthetix run');
 }
 
 export async function followerIsInstalled() {
   try {
-    await fs.access(path.join(ROOT, 'ipfs-cluster-follow/ipfs-cluster-follow'), fs.constants.F_OK);
+    await fs.access(
```
Comment (noisekit) [inline src/main/follower.ts:44]: IsWin check missing

File: src/main/follower.ts [hunk 2]
```diff
@@ -45,18 +56,26 @@ export async function followerDaemon() {
...
+  }
+
   if (!pid) {
     await configureFollower();
 
     try {
       // Cleanup locks in case of a previous crash
       await Promise.all([
-        fs.rm(path.join(IPFS_FOLLOW_PATH, 'synthetix/badger'), {
+        fs.rm(path.join(IPFS_FOLLOW_PATH, '.synthetix/badger'), {
```
Comment (noisekit) [inline src/main/follower.ts:74]: Revert this

File: src/main/follower.ts [hunk 3]
```diff
@@ -45,18 +56,26 @@ export async function followerDaemon() {
...
           recursive: true,
           force: true,
         }),
-        fs.rm(path.join(IPFS_FOLLOW_PATH, 'synthetix/api-socket'), {
+        fs.rm(path.join(IPFS_FOLLOW_PATH, '.synthetix/api-socket'), {
```
Comment (noisekit) [inline src/main/follower.ts:78]: And revert this

File: src/main/ipfs.ts [hunk 4]
```diff
@@ -35,7 +37,7 @@ export async function ipfsPid() {
 
 export async function ipfsIsInstalled() {
   try {
-    await fs.access(path.join(ROOT, 'go-ipfs/ipfs'), fs.constants.F_OK);
+    await fs.access(path.join(ROOT, 'go-ipfs/ipfs.exe'), fs.constants.F_OK);
```
Comment (noisekit) [inline src/main/ipfs.ts:40]: IsWin missing

File: src/main/follower.ts [hunk 5]
```diff
@@ -11,14 +11,16 @@ import logger from 'electron-log';
...
+import { getPlatformDetails } from './util';
 
 // Change if we ever want to store all follower info in a custom folder
 const HOME = os.homedir();
 const IPFS_FOLLOW_PATH = path.join(HOME, '.ipfs-cluster-follow');
 
 export function followerKill() {
   try {
-    getPidsSync('.synthetix/ipfs-cluster-follow/ipfs-cluster-follow').forEach((pid) => {
+    getPidsSync('.synthetix/ipfs-cluster-follow/ipfs-cluster-follow.exe').forEach((pid) => {
```
Comment (noisekit) [inline src/main/follower.ts:23]: IsWin check

File: src/main/ipfs.ts [hunk 6]
```diff
@@ -47,7 +49,15 @@ export async function ipfsDaemon() {
   if (!isInstalled) {
     return;
   }
-  const pid = await getPid('.synthetix/go-ipfs/ipfs daemon');
+
+  const isWin = os.platform() === 'win32';
+  let pid;
+  if (isWin) {
+    pid = await getPid('ipfs daemon');
```
Comment (noisekit) [inline src/main/ipfs.ts:56]: Need to check not only for Executable name but also for synthetix path so we don't mixup our ipfs daemon with potentially running another global one (in other places too)

File: src/main/follower.ts (hunk 1 above)
Comment (vderunov) [reply src/main/follower.ts:46]: Platform check (isWin) added

File: src/main/follower.ts (hunk 2 above)
Comment (vderunov) [reply src/main/follower.ts]: reverted

File: src/main/follower.ts (hunk 3 above)
Comment (vderunov) [reply src/main/follower.ts]: reverted

File: src/main/follower.ts (hunk 5 above)
Comment (vderunov) [reply src/main/follower.ts]: Platform check added

File: src/main/follower.ts [hunk 7]
```diff
@@ -140,18 +163,18 @@ export async function downloadFollower(_e?: IpcMainInvokeEvent, { log = logger.l
...
-    const file = createWriteStream(path.join(ROOT, 'ipfs-cluster-follow.tar.gz'));
+    const file = createWriteStream(path.join(ROOT, `ipfs-cluster-follow.${fileExt}`));
     https.get(downloadUrl, (response) => pipeline(response, file).then(resolve).catch(reject));
   });
 
   await new Promise((resolve, reject) => {
-    createReadStream(path.join(ROOT, 'ipfs-cluster-follow.tar.gz'))
-      .pipe(zlib.createGunzip())
+    createReadStream(path.join(ROOT, `ipfs-cluster-follow.${fileExt}`))
+      .pipe(fileExt === 'zip' ? unzipper.Extract({ path: ROOT }) : zlib.createGunzip())
```
Comment (noisekit) [inline src/main/follower.ts:177]: No need for separate lib, we can use https://nodejs.org/api/zlib.html, it has unzip too.

File: src/main/ipfs.ts (hunk 6 above)
Comment (vderunov) [reply src/main/ipfs.ts]: agreed to receive the process identifier (PID) of a running IPFS process by searching for a process named 'ipfs.exe'.

Comment (noisekit) [reply src/main/ipfs.ts]: Yeah we vcan check for more details with sone extra calls, but currently we cannot get i fo from tasklist anyway 🤷‍♂️

File: src/main/follower.ts [hunk 8]
```diff
@@ -140,18 +163,18 @@ export async function downloadFollower(_e?: IpcMainInvokeEvent, { log = logger.l
...
       .pipe(tar.extract({ cwd: ROOT }))
```
Comment (noisekit) [inline src/main/follower.ts:178]: I wonder if unzipper needs this? Suspect that it does not.

File: src/main/ipfs.ts [hunk 9]
```diff
@@ -124,19 +134,23 @@ export async function downloadIpfs(_e?: IpcMainInvokeEvent, { log = logger.log }
     log(`Installing ipfs version ${latestVersionNumber}`);
   }
 
-  const downloadUrl = `https://dist.ipfs.tech/go-ipfs/${latestVersion}/go-ipfs_${latestVersion}_darwin-${targetArch}.tar.gz`;
+  const downloadUrl = `https://dist.ipfs.tech/go-ipfs/${latestVersion}/go-ipfs_${latestVersion}_${osPlatform}-${targetArch}.${fileExt}`;
   log(`IPFS package: ${downloadUrl}`);
 
-  await fs.rm(path.join(IPFS_PATH, 'config'), { recursive: true });
+  const configPath = path.join(IPFS_PATH, 'config');
+  if (existsSync(configPath)) {
```
[the remark this answers is not in the document]
Comment (vderunov) [reply src/main/ipfs.ts]: Remove redundant file existence check before deletion

File: src/main/follower.ts (hunk 8 above)
Comment (vderunov) [reply src/main/follower.ts:178]: agreed to leave it as is

**Review by noisekit — APPROVED** [review]

**Pull request merged**